### PR TITLE
FND-307 - Eliminate unnecessary unions in restrictions in the Occurrences ontology

### PR DESCRIPTION
**Pull request by ElisaKendall**
Signed-off-by: Elisa Kendall <ekendall@thematix.com>

## Description

Eliminate unnecessary unions in restrictions in occurrences; note that there is one remaining union in a restriction in this ontology, on the occurrence class, which is definitional and thus retained.

Fixes: #1084 / FND-307; also relates to #1043 

## Checklist:

- [x] I'm familiar with the [FIBO developer quide](../CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [x] My contribution follows the [principles of best practices for FIBO](../ONTOLOGY_GUIDE.md).
- [x] My changes have been reconciled with latest master and no merge conflicts remain.
- [x] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [x] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [x] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

File: FND/DatesAndTimes/Occurrences.rdf [hunk 1]
```diff
@@ -77,39 +77,35 @@ They are modularized this way to minimize the ontological committments that are
...
-			</owl:Class>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-dt-oc;hasOccurrence"/>
+				<owl:onClass rdf:resource="&fibo-fnd-dt-oc;Occurrence"/>
+				<owl:minQualifiedCardinality rdf:datatype="&xsd;nonNegativeInteger">0</owl:minQualifiedCardinality>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-rel-rel;comprises"/>
```
Comment (rivettp) [inline FND/DatesAndTimes/Occurrences.rdf:88]: "comprises" seems a very odd property to use here: in no way is the Kind a part of the ScheduleEntry. 

However is it not the Occurrence rather than the ScheduleEntry that has the Kind? And that would be a restriction on the Occurrence - which we already have. Though that again uses an odd property "exemplifies". I would have thought "isClassifiedBy" would be better.
Or is this an attempt to *constrain* all the Occurrences in a given schedule to have the same Kind? If so that should be a restriction on the Schedule class with a cascaded restriction used (on the members of the Schedule). 

Bottom line - I propose to remove the Kind restriction on this, and all other xScheduleEntry classes, and replace the one on Occurrence to use isClassifiedBy instead of exemplifies.

Comment (ElisaKendall) [reply FND/DatesAndTimes/Occurrences.rdf:88]: So, the only change I made was to eliminate the union, I didn't change the restrictions themselves from what Mark created originally.  The intent, I believe, was to ensure that for a given regular schedule, there is consistency in the kinds of events that occur on a regular basis - the definition of 'regular'.  For example, a regular schedule of payments would have all and only occurrence kinds that are payments.

It's optional at the highest level, so that a scheduler could decide whether a given regular or ad hoc schedule includes only one kind of event, for example, a payment for a fixed rate loan, or more than one kind of event, say, a payment and a rate reset for a variable rate loan.  A given schedule entry should reflect only one kind of occurrence, though.  There are other issues with the restrictions on schedule entry, though - it inherits a combined date and time from dated collection constituent, which seems appropriate and should not be optional, but there are other optional restrictions at the higher level on date or date time that I think are unnecessary and lilkely confusing, so I'll fix that.

To retain the idea that Mark incorporated, that a regular schedule consists of events of a specific kind, or one or more specific kinds, seems important.  Whether comprises is the right verb for that is another matter, but comprises is a synonom of consists of, so I'm hesitant to remove it altogether.  A payment schedule wouldn't include events such as trades or settlements, only payments and possibly rate resets or rate assessments of some pre-determined kind.
